icu4j/main/core/src/main/java/com/ibm/icu/segmenter/RuleBasedSegmenter.java

```diff
@@ -0,0 +1,137 @@
+// © 2025 and later: Unicode, Inc. and others.
+// License & terms of use: https://www.unicode.org/copyright.html
+
+package com.ibm.icu.segmenter;
+
+import com.ibm.icu.text.BreakIterator;
+import com.ibm.icu.text.RuleBasedBreakIterator;
+import java.io.InputStream;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Performs segmentation according to the provided rule string. The rule string must follow the
+ * same guidelines as for {@link RuleBasedBreakIterator#getInstanceFromCompiledRules(InputStream)}.
+ * @draft ICU 78
+ */
+public class RuleBasedSegmenter implements Segmenter {
+
+  private String rules;
+
+  /**
+   * @draft ICU 78
+   */
+  @Override
+  public Segments segment(CharSequence s) {
+    return new RuleBasedSegments(s, this);
+  }
+
+  /**
+   * @return a builder for constructing {@code RuleBasedSegmenter}
+   * @draft ICU 78
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  RuleBasedSegmenter(String rules) {
+    this.rules = rules;
+  }
+
+  /**
+   * @internal
+   * @deprecated This API is ICU internal only.
+   */
+  @Override
+  @Deprecated
+  public BreakIterator getNewBreakIterator() {
+    return new RuleBasedBreakIterator(this.rules);
+  }
+
+  /**
+   * Builder for {@link RuleBasedSegmenter}
+   * @draft ICU 78
+   */
+  public static class Builder {
+
+    String rules;
+
+    Builder() { }
+
+    /**
+     * Sets the rule string for segmentation.
+     * @param rules rule string.  The rule string must follow the same guidelines as for
+     *     {@link RuleBasedBreakIterator#getInstanceFromCompiledRules(InputStream)}.
+     * @draft ICU 78
+     */
+    public Builder setRules(String rules) {
+      this.rules = rules;
+      return this;
+    }
+
+    /**
+     * Builds the {@code Segmenter}
+     * @return the constructed {@code Segmenter} instance
+     * @draft ICU 78
+     */
+    public Segmenter build() {
+      return new RuleBasedSegmenter(this.rules);
```

Comment:
This defers parsing/building the rules too much. If the rules have problems, we should throw an exception right here (or in the RuleBasedSegmenter constructor).

The RuleBasedSegmenter should have a field for the RBBI, not for the rule string. getNewBreakIterator() should clone the prototype.

Reply:
Done.